mautic/mautic
# DPMMA-2894 grapesjs asset manager lazy loading

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🟢
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
The current GrapesJS asset manager loads all media files at once, causing performance issues with large libraries.

This PR introduces:
- Lazy loading of files as the user scrolls.
- Ordering files by modification date, showing the most recent files first.







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Navigate to the GrapesJS editor in Mautic (e.g., create a new email or landing page) and click the **Builder** button
3. Open the Asset Manager by dragging the Image block to the editor
4. Observe the initial load of assets:
   - Only a limited number of assets should be visible at first
   - Assets should be ordered with the most recently modified files at the top
5. Scroll down in the Asset Manager:
   - New assets should load as you scroll
6. Upload a new asset:
   - It should appear at the top of the list after refreshing the Asset Manager
7. Test with a large number of assets:
   - If possible, test with an instance that has 100+ media files
   - Verify that the initial load is quick and scrolling remains smooth
8. Check browser console:
   - There should be no errors related to asset loading
9. Test on different browsers (Chrome, Firefox, Safari) to ensure consistent behavior


### :warning: DEPRECATIONS:
1. Method `FileManager::getImages` deprecated since Mautic 5.2, to be removed in 6.0. Use `FileManager::getImagePages` instead
2. Method `FileManagerController::assetsAction` deprecated since Mautic 5.2, to be removed in 6.0. Use `FileManagerController::getMediaAction` instead
3. Endpoint `GET /grapesjsbuilder/assets` deprecated since Mautic 5.2, to be removed in 6.0. Use `GET /grapesjsbuilder/media` instead
